src/behavior_oracle.cpp

```diff
@@ -0,0 +1,138 @@
+#include "behavior_oracle.h"
+
+#include "behavior.h"
+#include "bodypart.h"
+#include "player.h"
+#include "weather.h"
+
+#include <functional>
+
+using namespace behavior;
+
+status_t oracle_t::return_running() const
+{
+    return running;
+}
+
+// To avoid a local minima when the character has access to warmth in a shelter but gets cold
+// when they go outside, this method needs to only alert when travel time to known shelter
+// approaches time to freeze.
+status_t character_oracle_t::needs_warmth_badly() const
+{
+    const player *p = dynamic_cast<const player *>( subject );
+    // Use player::temp_conv to predict whether the Character is "in trouble".
+    for( const body_part bp : all_body_parts ) {
+        if( p->temp_conv[ bp ] <= BODYTEMP_VERY_COLD ) {
+            return running;
+        }
+    }
+    return success;
+}
+
+status_t character_oracle_t::needs_water_badly() const
+{
+    // Check thirst threshold.
+    if( subject->get_thirst() < 520 ) {
+        return running;
+    }
+    return success;
+}
+
+status_t character_oracle_t::needs_food_badly() const
+{
+    // Check hunger threshold.
+    if( subject->get_hunger() >= 300 && subject->get_starvation() > 2500 ) {
```

Comment:
I'm not sure which of needs_warmth, needs_water, or needs_food is coded correctly and which one is wrong, and maybe they are all correct, but it seems weird that needs_water is running when thirst is less than a threshold but needs_food is running when hunger is more than a threshold.

Reply:
and I'll be able to sort this out in the process of reigning in the scope of deciding to eat/drink.